resources/ApprovePage.js

```diff
@@ -7,9 +7,9 @@ const sendApprovalRequest = ( approve ) => {
 		.then( ( data ) => {
 			handleApprovalResponse( approve, data );
 		} )
-		.catch( error => {
+		.catch( ( error, data ) => {
 			console.error( 'API request failed:', error );
-			mw.notify( `API request failed: ${ error }`, { type: 'error' } );
+			mw.notify( data.xhr.responseJSON.message || 'API request failed: ${ error }', { type: 'error' } );
```

Comment:
Does this include things like authorization errors and the "revision id mismatch"? If so, the messaging should not start with "API request failed", as this isn't something non-technical users of the software understand.

Reply:
The human-readable message is used for known/our responses: `data.xhr.responseJSON.message`

If the API blows up in an unknown way then it's going to show that "API request failed:" message.